api/handlers/v1/resource.go

```diff
@@ -147,6 +147,35 @@ func (server APIServer) DeleteResource(ctx context.Context, request *entropyv1be
 	return &response, nil
 }
 
+func (server APIServer) ApplyAction(ctx context.Context, request *entropyv1beta1.ApplyActionRequest) (*entropyv1beta1.ApplyActionResponse, error) {
+	res, err := server.resourceService.GetResource(ctx, request.GetUrn())
+	if err != nil {
+		if errors.Is(err, store.ResourceNotFoundError) {
+			return nil, status.Error(codes.NotFound, "could not find resource with given urn")
+		}
+		return nil, status.Error(codes.Internal, "failed to apply fetch resource from db")
```

Comment:
before returning back resp to the client, how about logging the error as well ?

Reply:
Yes ideally, every internal server error should be logged out.